### ISSUE TITLE
String parameters for TerminationCharacter and LeadingSize should still use the sizeInBits as the maximum length of the string

### ISSUE THREAD

**Issue by xpromache** [yamcs/yamcs#2]
In the DataEncodingProcessor, when extracting String parameters:
switch(sde.getSizeType()) {
case Fixed:
sizeInBytes=sde.getSizeInBits()/8;
break;
case LeadingSize:
for(int i=0;i<sde.getSizeInBitsOfSizeTag();i+=8) {
sizeInBytes=(sizeInBytes<<8) + pcontext.bb.get(pcontext.bitPosition/8);
pcontext.bitPosition+=8;
}
break;
case TerminationChar:
int byteOffset=pcontext.bitPosition/8;
while(pcontext.bb.get(byteOffset)!=sde.getTerminationChar()){ 
sizeInBytes++;
byteOffset++;
}
break;
}

Basically in the last two cases it can read up to and more than the end of the buffer.
We should have a check if sde.getSizeInBits() returns <=0 then keep the actual behavior, if it returns something bigger than zero, that should be used as a maximum.

**Comment by xpromache** [comment]
This behaviour cannot be specified using XTCE, so closing for now.
